### ISSUE TITLE
Populate a prop from l3keys: ".prop_put:Nn"

### ISSUE THREAD

**Issue by blefloch** [latex3/latex3#444]
I suggest adding a key type to `l3keys`, like `.clilst_set:N` etc.
```
\keys_define:nn { mymod }
  { mykey .prop_put:Nn = \l_mypkg_prop { mykey } }
```

**Comment by eg9** [comment]
> On 24 Feb 2018, at 23:04 , Bruno Le Floch <notifications@github.com> wrote:
> 
> I suggest adding a key type to l3keys, like .clilst_set:N etc.
> 
> \keys_define:nn { mymod }
>   { mykey .prop_put:Nn = \l_mypkg_prop { mykey } }
Not sure about .clist_set:N, but it could be for uniformity. about .prop_put:Nn I definitely
agree: after all there’s big similarity between key-value and property list.

Ciao
Enrico

**Comment by blefloch** [comment]
(I didn't actually check the syntax.) In fact, `.clist_set:N` already exists.

**Comment by josephwright** [comment]
Do we want `.prop_put:N`, and take make the key name in the `prop` the same as the keyval, or do we want `.prop_put:Nn` with potentially differnt key names?

**Comment by eg9** [comment]
@josephwright Probably `mykey .prop_put:N = \l_module_task_prop` that does `\prop_put:Nnn \l_module_task_prop {mykey} { #1 }`

I don't see other uses and the syntax would be awkward anyway. For more complex tasks there's always `.code:n`

**Comment by wspr** [comment]
I agree with Enrico, with the possible addition it might be good to have something like

    .prop_prefix:n = {foo-}

to end up with result

    mykey .prop_put:N = \l_module_task_prop

becomes equivalent to

    \prop_put:Nnn \l_module_task_prop {foo-mykey} { #1 }

I’m not 100% sure about this, it might be better to keep it simple and park this idea ...
